GillianCore/utils/syntaxes.ml

```diff
@@ -6,4 +6,5 @@ end
 module Option = struct
   let ( let+ ) f o = Option.map o f
   let ( let* ) o f = Option.bind o f
+  let ( let- ) f o = Option.iter o f
```

Comment:
Yep, you figured out that wouldn't work I guess: iter doesn't compose like map and bind do.
bind is `'a t -> ('a -> 'b t) -> 'b t`
map is `'a t -> ('a -> 'b) -> 'b t`
Both take and return a `_ t` so you can compose those as you want.
iter is `'a t -> ('a -> ()) -> ()`, that doesn't compose, so you can't use monadic let binding.

Anyway, let's remove it :)

Reply:
Removed in b77c1f2afb712b4c68107ea9cf97b96714c4a3a2